File: backend/internal/domain/disk_stat.go
package domain

import "fmt"

type DiskStat struct {
	Path        string  `json:"path"`
	Fstype      string  `json:"fstype"`
	Total       uint64  `json:"total"`
	Free        uint64  `json:"free"`
	Used        uint64  `json:"used"`
	UsedPercent float64 `json:"usedPercent"`
}

func NewDiskStat(path, fstype string, total, free, used uint64, usedPercent float64) *DiskStat {
	return &DiskStat{
		Path:        path,
		Fstype:      fstype,
		Total:       total,
		Free:        free,
		Used:        used,
		UsedPercent: usedPercent,
	}
}

func (d DiskStat) ToString() string {
	return fmt.Sprintf(
		"Path: %s\nFstype: %s\nTotal: %d\nFree: %d\nUsed: %d\nUsedPercent: %f\n",
		d.Path, d.Fstype, d.Total, d.Free, d.Used, d.UsedPercent)
}

File: frontend/lib/models/disk_stat.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'disk_stat.freezed.dart';
part 'disk_stat.g.dart';

@freezed
class DiskStat with _$DiskStat {
  const factory DiskStat({
    required String path,
    required String fstype,
    required double total,
    required double free,
    required double used,
    @JsonKey(name: 'used_percent') required double usedPercent,
  }) = _DiskStat;

  factory DiskStat.fromJson(Map<String, dynamic> json) =>
      _$DiskStatFromJson(json);
}
